### PR TITLE
Refactor clasp pull

### PR DESCRIPTION
**Pull request by campionfellin**
This begins a larger effort of refactoring the `commands` code for better organization. 

Signed-off-by: campionfellin <campionfellin@gmail.com>

Begins work on https://github.com/google/clasp/issues/501

- [x] `npm run test` succeeds.
- [x] `npm run lint` succeeds.
- [ ] Appropriate changes to README are included in PR.

### CODE REVIEW

File: src/commands/pull.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import {
+    checkIfOnline,
```
Comment (grant) [inline src/commands/pull.ts]: 2 sp

File: src/commands/pull.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    fetchProject,
+    writeProjectFiles,
+} from './../files';
+
+/**
+ * Force downloads all Apps Script project files into the local filesystem.
+ * @param cmd.versionNumber {number} The version number of the project to retrieve.
+ *                             If not provided, the project's HEAD version is returned.
+ */
+export const pull = async (cmd: { versionNumber: number }) => {
```
Comment (grant) [inline src/commands/pull.ts]: Maybe export default if we are going with 1 command per file?

```ts
export default async (cmd: { versionNumber: number }) => {
```

File: src/index.ts [hunk 3]
```diff
@@ -47,6 +46,10 @@ import {
 } from './commands';
 import { PROJECT_NAME, handleError } from './utils';
 
+import {
```
Comment (grant) [inline src/index.ts]: After _default_ exports:
```ts
import pull from './commands/pull';
```

**Comment by grant** [comment]
Thanks!
Please add an issue # with all PRs.
Create an issue if there isn't one.

**Comment by campionfellin** [comment]
@grant sure, though I think it's hard to classify this as an "issue". Maybe more of a "project" (https://github.com/google/clasp/projects) with tasks for each command?

File: src/commands/pull.ts (hunk 2 above)
Comment (campionfellin) [reply src/commands/pull.ts]: Great idea

**Comment by grant** [comment]
You can have multiple PRs for 1 issue, just make it to track the issue.

imo better code organization won't help much for most users, it may be better to focus on better TypeScript integration or better `clasp run`.

Congrats on #500 btw.

**Comment by campionfellin** [comment]
Ok, I'll go with an issue. I think better organization will help reduce tech debt and make it easier to improve `clasp run` (function currently at ~150 lines).

And thanks :) actually was not planned, just finally had a bit of free time.

**Comment by grant** [comment]
LGTM, thanks!

**Pull request merged**